salt/user-certs/init.sls

```diff
@@ -0,0 +1,44 @@
+include:
+  - crypto
+
+{% for cert in ['velum', 'salt-api'] %}
+/etc/pki/{{ cert }}.key:
+  x509.private_key_managed:
+    - bits: 4096
+    - require:
+      - sls:  crypto
+      - file: /etc/pki
+  file.managed:
+    - replace: false
+    - user: root
+    - group: root
+    - mode: 644
+
+/etc/pki/{{ cert }}.crt:
+  x509.certificate_managed:
+    - ca_server: {{ salt['mine.get']('roles:ca', 'x509.get_pem_entries', expr_form='grain').keys()[0] }}
+    - signing_policy: minion
+    - public_key: /etc/pki/{{ cert }}.key
+    - CN: {{ grains['fqdn'] }}
```

Comment:
Isn't the risk of picking a bad value here? However I don't think there's much we can do...